doc/source/building_images/build_expandable_disk.rst

```diff
@@ -393,6 +393,15 @@ oemconfig.oem-unattended Element
   possible target devices are discovered the image is deployed to
   the first device. ``kiwi_oemunattended`` in the initrd
 
+oemconfig.oem-unattended-id Element
+  The target disk device for the installation is selected according
+  to the specified device ID. The device ID corresponds to the name of
+  the device as it exists for the configured `devicepersistency`. By
+  default this is the `by-uuid` device name. If no such representation
+  exist e.g for ramdisk devices, the unix device node can be used
+  to select. The given name must be present in the device list detected
+  by kiwi.
+
```

Comment:
Added documentation, nice :+1:

 Probably we should note that  `oem-unattended-id` has preference over `rd.kiwi.oem.installdevice` (or the other way round if you think it should be the opposite)

Reply:
> I understand that oem-unattended-id element from the XML will always have precedence over rd.kiwi.oem.installdevice

I believe this is not the case. From a code logic we only evaluate kiwi_oemunattended_id if there is more than one device found. As we set the list of devices to exactly one in the rd.kiwi.oem.installdevice case we do not enter that code path.

But I agree to you this should be done more explicit. I suggest to reset any eventually given kiwi_oemunattended_id and also kiwi_oemdevicefilter in case rd.kiwi.oem.installdevice is set.

Reply:
@davidcassany I added a commit which addresses your issue. I think we want that rd.kiwi.oem.installdevice as a kernel parameter takes precedence over any other setting and the docs should also be clear on that. What do you think ?

Reply:
> I believe this is not the case. From a code logic we only evaluate kiwi_oemunattended_id if there is more than one device found. As we set the list of devices to exactly one in the rd.kiwi.oem.installdevice case we do not enter that code path.

Oh, indeed, you are right. Then from a code perspective also setting `kiwi_oemunattended` to `true` had no real effect, I guess that's why I got confused as I directly looked when this variable was used without reading the full code 😅 

Yes the current approach looks all good to me. Resetting all overwritten (or ignored) parameters looks like the most readable and maintainable approach :+1:.

Reply:
>Then from a code perspective also setting kiwi_oemunattended to true had no real effect
Well it had one at the code path that asks to continue later
```bash
if [ "${kiwi_oemunattended}" = "false" ];then
        local ack_dump_text="Destroying ALL data on ${image_target}, continue ?"
        ...
fi
```
So this setting was indeed needed :) For all the others you are right, but now it's explicit and better to read and understand. Thanks for the feedback and review :+1: